File: 1.3-URLify/URLify.js
// Write a method to replace all spaces in a string with '%20'. You may assume that the string has sufficient space at the end to hold the additional characters, and that you are given the "true" length of the string.

const URLify = (str) => {
  str = str.split('')
  for (let i = 0; i < str.length; i++) {
    if (str[i] === ' ') {
      str[i] = '%20'
    }
  }
  return str.join('')
}

console.log(URLify('testing one two'))
console.log(URLify('nothing-changed'))
console.log(URLify(' one last one to test '))
console.log(URLify(' '))

File: 1.3-URLify/URLify.rb
# Write a method to replace all spaces in a string with '%20'. You may assume that the string has sufficient space at the end to hold the additional characters, and that you are given the "true" length of the string.

def URLify(str)
  str = str.split(//)
  str.each_with_index do |char, i|
    if char == ' '
      str[i] = '%20'
    end
  end
  return str.join()
end

puts URLify('testing one two')
puts URLify('nothing-changed')
puts URLify(' one last one to test ')
puts URLify(' ')